packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -409,100 +437,158 @@ function useMediaUploadQueue() {
+     * is being optimized/muted/trimmed, which will cause
+     * a new resource to be created and uploaded.
+     *
+     * As long as that resource's upload hasn't been cancelled,
+     * it is considered to be processing.
+     *
+     * @param {number} resourceId Resource ID.
+     * @return {boolean} Whether the resource is processing.
      */
     const isResourceProcessing = (resourceId) =>
       state.queue.some(
         (item) =>
-          [
-            ITEM_STATUS.PENDING,
-            ITEM_STATUS.UPLOADING,
-            ITEM_STATUS.TRANSCODING,
-            ITEM_STATUS.MUTING,
-            ITEM_STATUS.TRIMMING,
-          ].includes(item.state) && item.originalResourceId === resourceId
+          item.state !== ITEM_STATUS.CANCELLED &&
+          item.originalResourceId === resourceId
       );
+
     /**
-     * Is the current resource being processed.
+     * Determine whether the current resource is being processed.
      *
-     * @param {number} resourceId Resource id.
-     * @return {boolean} if resource with id is found.
+     * This is the case when uploading a new media item that first
+     * needs to be transcoded/muted/trimmed.
+     *
+     * As long as that resource's upload hasn't been cancelled,
+     * it is considered to be processing.
+     *
+     * @param {number} resourceId Resource ID.
+     * @return {boolean} Whether the resource is processing.
      */
     const isCurrentResourceProcessing = (resourceId) =>
       state.queue.some(
         (item) =>
-          [
-            ITEM_STATUS.PENDING,
-            ITEM_STATUS.UPLOADING,
-            ITEM_STATUS.TRANSCODING,
-            ITEM_STATUS.MUTING,
-            ITEM_STATUS.TRIMMING,
-          ].includes(item.state) && item.resource.id === resourceId
+          item.state !== ITEM_STATUS.CANCELLED &&
+          (item.resource.id === resourceId ||
+            item.previousResourceId === resourceId)
       );
+
     /**
-     * Is the current resource uploading.
+     * Determine whether the current resource is being uploaded.
      *
-     * @param {number} resourceId Resource id.
-     * @return {boolean} if resource with id is found.
+     * This is the case when uploading a new media item after any initial
+     * transcoding/muting/trimming has already happened.
+     *
+     * Checks for both `resource.id` as well as `previousResourceId`,
+     * since after upload to the backend, the resource's temporary uuid
+     * will be replaced by the permanent ID from the backend.
+     *
+     * @param {number} resourceId Resource ID.
+     * @return {boolean} Whether the resource is uploading.
      */
     const isCurrentResourceUploading = (resourceId) =>
       state.queue.some(
         (item) =>
-          item.state === ITEM_STATUS.UPLOADING &&
-          item.resource.id === resourceId
+          [
+            ITEM_STATUS.PENDING,
+            ITEM_STATUS.UPLOADING,
+            ITEM_STATUS.UPLOADED,
```

Comment:
Naming of this function seemed wrong now? Why is this change needed?

Reply:
I'll check if more tweaks are needed, but this comparison allows checking whether the media item is being uploaded (or preparing to be uploaded), but not currently being transcoded/trimmed/muted.

"Uploaded" still means there's some unfinished business, so that's included here as well.

Reply:
Would be nice to have it written out somewhere what each status means and in which order things happen.

Does `UPLOADED` mean that it goes back to `PENDING` since it's waiting for new unfinished business to be completed now? Or is the next step happening out of the upload queue and thus not `PENDING`, e.g. just waiting to replace the origin resource? Is there another state after `UPLOADED`?

Reply:
I suppose I might want to extend my diagram.

> Does UPLOADED mean that it goes back to PENDING since it's waiting for new unfinished business to be completed now?

No. States never occur twice. `PENDING`  is only at the very beginning before anything happens.

> Or is the next step happening out of the upload queue and thus not PENDING, e.g. just waiting to replace the origin resource?

Correct

>  Is there another state after UPLOADED?

No

----

It's basically:

`PENDING`  -> `TRANSCODING` -> `TRANSCODED` -> `TRIMMING` -> `TRIMMED` -> `MUTING` -> `MUTED` -> `UPLOADING` -> `UPLOADED`

transcoding/trimming/muting steps are of course optional, depending on what has been passed.

Reply:
Maybe we could add documentation in this [file](https://github.com/google/web-stories-wp/blob/d8c96c5ae1122cdd9210cf559c4aac56a7f2e1b2/packages/story-editor/src/app/media/utils/useMediaUploadQueue/constants.js).

Reply:
Add some documentation on top of the `useMediaUploadQueue` hook